libraries/AP_Networking/AP_Networking_PPP.cpp

```diff
@@ -0,0 +1,197 @@
+        msg = "PPPERR_OPEN";
+        break;
+    }
+    case PPPERR_DEVICE: {          /* Invalid I/O device for PPP. */
+        msg = "PPPERR_DEVICE";
+        break;
+    }
+    case PPPERR_ALLOC: {           /* Unable to allocate resources. */
+        msg = "PPPERR_ALLOC";
+        break;
+    }
+    case PPPERR_USER: {            /* User interrupt. */
+        msg = "PPPERR_USER";
+        break;
+    }
+    case PPPERR_CONNECT: {         /* Connection lost. */
+        msg = "PPPERR_CONNECT";
+        break;
+    }
+    case PPPERR_AUTHFAIL: {        /* Failed authentication challenge. */
+        msg = "PPPERR_AUTHFAIL";
+        break;
+    }
+    case PPPERR_PROTOCOL: {        /* Failed to meet protocol. */
+        msg = "PPPERR_PROTOCOL";
+        break;
+    }
+    case PPPERR_PEERDEAD: {        /* Connection timeout */
+        msg = "PPPERR_PEERDEAD";
+        break;
+    }
+    case PPPERR_IDLETIMEOUT: {     /* Idle Timeout */
+        msg = "PPPERR_IDLETIMEOUT";
+        break;
+    }
+    case PPPERR_CONNECTTIME: {     /* Max connect time reached */
+        msg = "PPPERR_CONNECTTIME";
+        break;
+    }
+    case PPPERR_LOOPBACK: {        /* Loopback detected */
+        msg = "PPPERR_LOOPBACK";
+        break;
+    }
+    }
+    if (msg != nullptr) {
+        GCS_SEND_TEXT(MAV_SEVERITY_INFO, "PPP: %s", msg);
+    }
+}
+
+
+/*
+  initialise PPP network backend using LWIP
+ */
+bool AP_Networking_PPP::init()
+{
+    auto &sm = AP::serialmanager();
+    uart = sm.find_serial(AP_SerialManager::SerialProtocol_PPP, 0);
+    if (uart == nullptr) {
+        return false;
+    }
+
+    pppif = new netif;
+    if (pppif == nullptr) {
+        return false;
+    }
+
+    // initialise TCP/IP thread
+    LWIP_TCPIP_LOCK();
+    tcpip_init(NULL, NULL);
+    LWIP_TCPIP_UNLOCK();
+
+    hal.scheduler->delay(100);
+    
+    // create ppp connection
+    LWIP_TCPIP_LOCK();
+
+    ppp = pppos_create(pppif, ppp_output_cb, ppp_status_callback, this);
+    if (ppp == nullptr) {
+        GCS_SEND_TEXT(MAV_SEVERITY_INFO, "PPP: failed to create link");
+        return false;
+    }
+    LWIP_TCPIP_UNLOCK();
+
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO, "PPP: started");
+    hal.scheduler->thread_create(FUNCTOR_BIND_MEMBER(&AP_Networking_PPP::ppp_loop, void),
+                                 "ppp",
+                                 2048, AP_HAL::Scheduler::PRIORITY_NET, 0);
+
+    return true;
+}
+
+/*
+  main loop for PPP
+ */
+void AP_Networking_PPP::ppp_loop(void)
+{
+    while (!hal.scheduler->is_system_initialized()) {
+        hal.scheduler->delay_microseconds(1000);
+    }
+    
+    // connect and set as default route
+    LWIP_TCPIP_LOCK();
+    ppp_connect(ppp, 0);
+
+    netif_set_default(pppif);
+    LWIP_TCPIP_UNLOCK();
+
+    // ensure this thread owns the uart
+    uart->begin(0);
+    uart->set_unbuffered_writes(true);
+
+    while (true) {
+        uint8_t buf[1024];
```

Comment:
why 1024? Why not MTU?

Reply:
no particular reason, its PPP bytes, so not MTU related